crates/red_knot/src/program/check.rs

```diff
@@ -39,7 +39,30 @@ impl Program {
     ) -> Result<Diagnostics, CheckError> {
         context.cancelled_ok()?;
 
-        // TODO schedule the dependencies.
+        let symbol_table = self.symbol_table(file);
+        let dependencies = symbol_table.dependencies();
+
+        if !dependencies.is_empty() {
+            let module_name = self.file_to_module(file).map(|module| module.name(self));
+
+            // TODO scheduling all dependencies here is wasteful if we don't infer any types on them
+            //  but I think that's unlikely, so it is okay?
+            //  Anyway, we need to figure out a way to retrieve the dependencies of a module
+            //  from the persistent cache. So maybe it should be a separate query after all.
+            for dependency in dependencies {
+                let dependency_name = dependency.module_name(module_name.as_ref());
+
+                if let Some(dependency_name) = dependency_name {
+                    // TODO The resolver doesn't support native dependencies yet.
```

Comment:
I think this TODO is a little confusing, because we'll never support "native dependencies." What we'll support in the future is stub files, and stub files in typeshed. Which are used to provide type information both for native and non-native dependencies in the standard library. (And also for some third-party code, and sometimes first-party code too.)

Also this seems like an odd place for the TODO, because nothing here will need to change to support that.

```suggestion
                    // TODO The resolver doesn't support stub files or typeshed yet.
```